osu.Game/Screens/Edit/Compose/Components/DistanceSnapGrid.cs

```diff
@@ -146,5 +146,20 @@ protected ColourInfo GetColourForIndexFromPlacement(int placementIndex)
             int repeatIndex = placementIndex / beatDivisor.Value;
             return ColourInfo.SingleColour(colour).MultiplyAlpha(0.5f / (repeatIndex + 1));
         }
+
+        public float GetBeatSnapDistanceAt(HitObject referenceObject) => (float)(snapProvider.GetBeatSnapDistanceAt(referenceObject) * distanceSpacingMultiplier.Value);
+        public float DurationToDistance(HitObject referenceObject, double duration) => (float)(snapProvider.DurationToDistance(referenceObject, duration) * distanceSpacingMultiplier.Value);
+        public double DistanceToDuration(HitObject referenceObject, float distance) => snapProvider.DistanceToDuration(referenceObject, (float)(distance / distanceSpacingMultiplier.Value));
+        public double GetSnappedDurationFromDistance(HitObject referenceObject, float distance) => snapProvider.GetSnappedDurationFromDistance(referenceObject, (float)(distance / distanceSpacingMultiplier.Value));
+
+        public float GetSnappedDistanceFromDistance(HitObject referenceObject, float distance)
+        {
+            float snappedDistance = snapProvider.GetSnappedDistanceFromDistance(referenceObject, (float)(distance / distanceSpacingMultiplier.Value));
+            return (float)(snappedDistance * distanceSpacingMultiplier.Value);
+        }
```

Comment:
This plays around with math a bit, but the summary of it is that input distances need to have the multipler removed (e.g. `DistanceToDuration`, `GetSnappedDurationFromDistance`), while output distances need to have the multiplier applied (e.g. `DurationToDistance`). And then there's `GetSnappedDistanceFromDistance` which does both.

Not sure if I'm the only one who finds that not obvious, but I don't know how it can be explained in an inline comment either.